docs/Forms.md

```diff
@@ -0,0 +1,142 @@
+    },
+    {
+      "name": "buyerZip",
+      "constant": false,
+      "type": "text",
+      "value": null,
+      "required": false,
+      "label": "Postcode",
+      "helpText": null,
+      "fields": []
+    },
+    {
+      "name": "buyerState",
+      "constant": false,
+      "type": "text",
+      "value": null,
+      "required": false,
+      "label": "State",
+      "helpText": null,
+      "fields": []
+    },
+    {
+      "name": "buyerCountry",
+      "constant": false,
+      "type": "text",
+      "value": null,
+      "required": true,
+      "label": "Country",
+      "helpText": null,
+      "fields": []
+    }
+  ]
+}
+```
+
+Output:
+
+![Form](./img/Forms-1.png "Form")
+
+In a field definition, only the following fields can be set:
+
+| Field   |      Description      |
+|----------|-------------|
+`.fields.constant` | If `true`, the `.value` must be set in the form definition, and the user will not be able to change the field's value. ( example: the form definition's version) |
+| `.fields.type` | The HTML input type `text, radio, checkbox, password, hidden, button, color, date, datetime-local, month, week, time, email, number, range, search, url, select, tel` |
+| `.fields.options` | If `.fields.type` is `select`, the list of selectable values |
+| `.fields.options.text` | The text displayed for this option |
+| `.fields.options.value` | The value of the field if this option is selected |
+| `.fields.type=fieldset` | Create a HTML `fieldset` around the children `.fields.fields` (see below) |
+| `.fields.name` | The JSON property name of the field as it will appear in the invoice's metadata. |
+| `.fields.value` | The default value of the field |
+| `.fields.required` | if `true`, the field will be required |
+| `.fields.label` | The label of the field |
+| `.fields.helpText` | Additional text to provide an explanation for the field. |
+| `.fields.fields` | You can organize your fields in a hierarchy, allowing child fields to be nested within the invoice's metadata. This structure can help you better organize and manage the collected information, making it easier to access and interpret. For example, if you have a form that collects customer information, you can group the fields under a parent field called customer. Within this parent field, you might have child fields like name, email, and address.  |
+
+
+The values of the fields are stored in the metadata of the invoice.
+
+## Well-known field names
+
+The field name represents the JSON property name that stores the user-provided value in the invoice's metadata.
+
+Some well-known names can be interpreted and modify the invoice's settings.
+
+| Field name   |      Description      |
+|----------|-------------|
+| `invoice_amount` | The invoice's amount |
+| `invoice_currency` | The invoice's currency |
+
+## Pre-filling form values
+
+You can pre-fill the fields of an invoice automatically by adding query strings to the form's URL, such as `?your_field=value`.
```

Comment:
Please clarify if pre-filling works for fields that have been set to constant: true

At least thats what stays as an open question for me when reading this. 😄

Reply:
@petzsch `constant: true` doesn't have any interaction with pre-filling.
Constant means that whatever its value set in the form, it will always be ignored. Instead, the value set in the template is always the one chosen.